tracex_project/extraction/tests.py

```diff
@@ -0,0 +1,169 @@
+        """Tests that the Orchestrator's configuration remains unchanged with subsequent instantiations."""
+        Orchestrator.reset_instance()
+        config1 = ExtractionConfiguration()
+        orchestrator1 = Orchestrator(config1)
+        config2 = ExtractionConfiguration()
+        orchestrator2 = Orchestrator(config2)
+        self.assertIs(orchestrator1.configuration, orchestrator2.configuration)
+
+    # Does not work yet, need to adjust initialization of modules
+    def test_initialize_modules(self):
+        """Tests if initialize_modules correctly initializes all modules."""
+        Orchestrator.reset_instance()
+        config = MagicMock()
+
+        activity_labeling_mock = MagicMock()
+        activity_labeling_mock.name = "activity_labeling"
+        time_extraction_mock = MagicMock()
+        time_extraction_mock.name = "time_extraction"
+        event_type_classification_mock = MagicMock()
+        event_type_classification_mock.name = "event_type_classification"
+        location_extraction_mock = MagicMock()
+        location_extraction_mock.name = "location_extraction"
+
+        config.modules = {
+            "activity_labeling": MagicMock(return_value=activity_labeling_mock),
+            "time_extraction": MagicMock(return_value=time_extraction_mock),
+            "event_type_classification": MagicMock(return_value=event_type_classification_mock),
+            "location_extraction": MagicMock(return_value=location_extraction_mock),
+        }
+
+        orchestrator = Orchestrator(config)
+        modules = orchestrator.initialize_modules()
+
+        self.assertEqual(len(modules), 4)
+        self.assertEqual(modules[0].name, "activity_labeling")
+        self.assertEqual(modules[1].name, "time_extraction")
+        self.assertEqual(modules[2].name, "event_type_classification")
+        self.assertEqual(modules[3].name, "location_extraction")
+
+
+class ActivityLabelerTests(TestCase):
+    """Test cases for the ActivityLabeler."""
+
+    def test_execute_return_value(self):
+        """Tests if the return value of the execute method always is a dataframe and if column name is as expected."""
+        test_data = ["I fell ill yesterday.", "I went to the doctor today."]
+        activity_labeler = ActivityLabeler()
+        result = activity_labeler.execute(patient_journey_sentences=test_data)
+        self.assertIsInstance(result, pd.DataFrame)
+        self.assertIn("activity", result.columns)
+        self.assertIn("sentence_id", result.columns)
+
+
+class TimeExtractorTests(TestCase):
+    """Test cases for the TimeExtractor."""
+
+    def test_execute_return_value(self):
+        """Tests if the return value of the execute method is always a dataframe and if column names are as expected."""
+        data = {
+            'activity': ['fell ill'],
+            'sentence_id': ['1']
+        }
+        patient_journey = ["1: I fell ill on June 5 and recovered on June 7."]

```

Comment:
```suggestion
        patient_journey = ["I fell ill on June 5 and recovered on June 7."]
```

The "1: " shouldn't be there as in the data provided in the pipeline it's not there either.